ghga_datasteward_kit/file_ingest.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Interaction with file ingest service"""
+
+from pathlib import Path
+from typing import Callable
+
+import httpx
+from pydantic import BaseSettings, Field, ValidationError
+
+from ghga_datasteward_kit import models, utils
+
+
+class IngestConfig(BaseSettings):
+    """Config options for calling the file ingest endpoint"""
+
+    endpoint_base: str = Field(
+        ..., description="Base URL under which the /ingest endpoint is available."
+    )
+    pubkey: str = Field(
+        ..., description="Public key used for encryption of the payload."
+    )
+    token: str = Field(
+        ...,
+        description="Token used for authentication against the file ingest service.",
+    )
```

Comment:
Token doesn't need to be part of config but read in using the function from the utils package.